pygsti/circuits/circuit.py

```diff
@@ -965,21 +974,39 @@ def __pow__(self, x):  # same as __mul__()
         return self.__mul__(x)
 
     def __eq__(self, x):
-        if x is None: return False
+        
         if isinstance(x, Circuit):
-            return self.tup.__eq__(x.tup)
+            if len(self) != len(x):
+                return False
+            else:
+                if self._static and x._static:
+                    return self._hashable_tup == x._hashable_tup
```

Comment:
Can we not just check `_hash` directly in this case, since we store those?

Reply:
Great point! That is indeed faster.